src/libostree/ostree-repo-commit.c

```diff
@@ -668,6 +668,34 @@ _ostree_repo_bare_content_commit (OstreeRepo                 *self,
 {
   OstreeRealRepoBareContent *real = (OstreeRealRepoBareContent*) barewrite;
   g_assert (real->initialized);
+
+  if ((self->min_free_space_percent > 0 || self->min_free_space_mb > 0) && self->in_transaction)
+    {
+      struct stat st_buf;
+      if (!glnx_fstat (real->tmpf.fd, &st_buf, error))
+        return FALSE;
+
+      g_mutex_lock (&self->txn_lock);
+      g_assert_cmpint (self->txn.blocksize, >, 0);
+
+      const fsblkcnt_t object_blocks = (st_buf.st_size / self->txn.blocksize) + 1;
+      if (object_blocks > self->txn.max_blocks)
+        {
+          self->cleanup_stagedir = TRUE;
+          g_mutex_unlock (&self->txn_lock);
+          g_autofree char *formatted_required = g_format_size (st_buf.st_size);
```

Comment:
This whole block which checks for whether there’s enough free space, and errors if there isn’t, could be factored out into a helper function to avoid it being repeated 3 times.

Reply:
At the very least, the repetition of the same 2 error messages 3 times could be avoided.

Reply:
Sure, something like the following would make me happy:
```
static gboolean
throw_min_free_space_error (OstreeRepo  *self,
                            guint64      bytes_required,
                            GError     **error)
{
  g_autofree char *formatted_free = g_format_size (bytes_required);
  if (self->min_free_space_percent > 0)
    return glnx_throw (error, "min-free-space-percent '%u%%' would be exceeded, %s available",
                       self->min_free_space_percent, formatted_free);
  else
    return glnx_throw (error, "min-free-space-size %" G_GUINT64_FORMAT "MB would be exceeded, %s available",
                       self->min_free_space_mb, formatted_free);
}
```